docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,275 @@
+versioned at a different pace than Avocado.
+
+Every week a automatic job will be executed to determine if there is a
+potential need for a new release:
+
+* If the number of commits since last release has reached a threshold; AND
+* There is no open discussion with the tag “release-decision”:
+  * Creates a new discussion (with the "release-decision" tag) to run a poll
+    between all the MAINTAINERS, so they will have the opportunity to thumbs
+    up if there is a need for a new release;
+  * The link to the new discussion should be sent to all MAINTAINERS;
+  * If the decision is positive, a round-robin selected maintainer should be responsible for coordinating the release it self;
+  * And the discussion should be closed
+* If there is any existing open discussion, the bot job should comment on the
+  same discussion with the updated list of commits and ping again the
+  MAINTAINERS, give them the opportunity to thumbs up based on the new
+  status;
+
+Backwards Compatibility
+***********************
+
+The current libraries are not going to be removed right away. A transition
+period is expected for migrating each library into this new standard, and
+because of that libraries in transition should be available, receive bug fixes
+for at least one LTS cycle.
+
+Each library in our current code base, should receive a status flag, informing
+users what their current status is, according on the following table:
+
+1. maintainer-wanted
+
+  The library is using the legacy mode, and it is not yet converted to the
+  autils repository. This should be the status for all avocado.utils.*
+  libraries upon the creation of the autils repository.
+
+2. adopted
+
+  A maintainer has picked up this library to be ported to the autils
+  repository.  It’s not expected yet that a working version is available on
+  autils, but someone is working on the migration.  During this period, the
+  avocado.utils library should continue to receive both bug fixes and
+  improvements.  It’s the maintainer’s responsibility to make sure that the
+  work in progress version of the new autils library includes the same bug
+  fixes and improvements merged during this phase.
+
+3. migrated
+
+ This means the current avocado.utils library is deprecated in favor of the
+ migrated version at the autils repository. During at least one Avocado LTS
+ cycle period, the avocado.utils version of the libraries should:
+
+  * Include warnings in each deprecated method or function, pointing users to
+    the equivalent method or function in the new autils library
+  * Keep receiving critical bug fixes backported from the new autils version
+
+ At this point, this library has an official maintainer who will be the main
+ point of contact for.
+ 
+4. abandoned
+
+  * There are no plans to migrate the library to autils.
+
+Once at least one LTS cycle has passed after the “migrated” status, the legacy
+library can be permanently removed from avocado repository.
```

Comment:
Or Avocado VT repository or I guess any other repository where the library is contributed from since I didn't see a restriction requiring it to originate from `avocado.utils` only.

Reply:
I agree with the Avocado-VT part, but I guess the idea here it was to document only the parts that we control. We can't remove the library from "other repositories".

Reply:
> I agree with the Avocado-VT part, but I guess the idea here it was to document only the parts that we control. We can't remove the library from "other repositories".

Yes definitely so I guess a generic referral as "...the legacy library can be permanently removed from its repository of origin" sounds clearer regarding the scope of the new repo.